Source/WebGPU/WebGPU/RenderPipeline.mm

```diff
@@ -333,6 +322,17 @@ static MTLVertexStepFunction stepFunction(WGPUVertexStepMode stepMode)
     return vertexDescriptor;
 }
 
+static auto buildKeyValueReplacements(const auto& stage)
```

Comment:
I used `auto` here because `descriptor.vertex` and `fragmentDescriptor` are different types

Reply:
Does that mean we get two `buildKeyValueReplacements` functions overloaded on return type?